temporallib/auth/auth.py

```diff
@@ -14,51 +14,76 @@
 import google.auth.transport.requests
 from typing import Union
 from dataclasses import asdict
+import os
 
 
-@dataclass
 class MacaroonAuthOptions:
     """
     Defines the parameters for authenticating with Candid.
     """
 
-    macaroon_url: str
-    username: str
-    keys: KeyPair
+    macaroon_url: str = None
+    username: str = None
+    keys: KeyPair = None
+
+    def __init__(self, macaroon_url: str = None, username: str = None, keys: KeyPair = None):
+        self.macaroon_url = macaroon_url or os.getenv("TEMPORAL_CANDID_URL")
+        self.username = username or os.getenv("TEMPORAL_CANDID_USERNAME")
+        self.keys = keys or KeyPair()
+
 
 
 @dataclass
 class GoogleAuthOptions:
     """
     Defines the parameters for authenticating with Google IAM.
     """
+    type: str = None
+    project_id: str = None
+    private_key_id: str = None
+    private_key: str = None
+    client_email: str = None
+    client_id: str = None
+    auth_uri: str = None
+    token_uri: str = None
+    auth_provider_x509_cert_url: str = None
+    client_x509_cert_url: str = None
+
+    def __post_init__(self):
+        self.type = self.type or os.getenv("TEMPORAL_OIDC_AUTH_TYPE")
```

Comment:
suggestion: How about using something like:
```python
def set_env_attr(self, attr, env):
    if not self.getattr(attr):
        self.setattr(attr, os.getenv(env))
```
It might require some adjustment to make the snippet work.

You can also augment either this function or define another one to do this for dicts or lists that map attributes to environment variable names.